### PR TITLE
[16.0][MIG] storage_backend

### PR DESCRIPTION
**Pull request by HviorForgeFlow**
Standard migration

CC @ForgeFlow

### CODE REVIEW

**Comment by lmignon** [comment]
@HviorForgeFlow A complete rewrite of this addon has been discussed with @sebastienbeau  I don't know what's the status of this refactoring but the indent is to get rid of the dependency on component and to simplify the code.

**Comment by HviorForgeFlow** [comment]
OK @lmignon closing the PR until it gets clarified

**Comment by lmignon** [comment]
@sebastienbeau Can you give some information on the refactoring. Is-it still in progress, do you need some help?

**Comment by JordiBForgeFlow** [comment]
@lmignon @sebastienbeau we will maitain a workable local branch with the migrated modules, and we will then review your proposal once you submit

**Comment by simahawk** [comment]
> @HviorForgeFlow A complete rewrite of this addon has been discussed with @sebastienbeau I don't know what's the status of this refactoring but the indent is to get rid of the dependency on component and to simplify the code.

@lmignon @sebastienbeau would be nice to know more about this. I wouldn't remove `component` but I'm eager to hear about your motivations :)

**Comment by lmignon** [comment]
@simahawk The goal is to lower the complexity of this addon and improve the maintainability. Extensibility can be achieved trough a different approach than the one proposed by the component addon. The removing of the dependency on component is one part of the refactoring. I don't know where Sebastien stands on this issue. We had exchanges in September and October during which I had proposed a class diagram modelling a different approach.

**Comment by lmignon** [comment]
@HviorForgeFlow Can you reopen this PR to highlight the current discussion? (and prevent others PR for the migration) 
Thank you

**Comment by lmignon** [comment]
@HviorForgeFlow I propose to move forward with this migration... we can't block with a refactoring that could take too much time....

Can you fix the tests and finalize the migration?

**Comment by HviorForgeFlow** [comment]
@lmignon build is green now, having some issues on next module migrations, but this one at least is passing the tests.

**Review by lmignon — APPROVED** [review]
Thank you @HviorForgeFlow

**Review by anothingguy — APPROVED** [review]

File: test-requirements.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+mock
```
Comment (simahawk) [inline test-requirements.txt:1]: I don't understand this change. `mock` is part of `unittest` now...

Comment (HviorForgeFlow) [reply test-requirements.txt:1]: Pipeline was failing: https://github.com/OCA/storage/actions/runs/4163534197/jobs/7204052623

Comment (simahawk) [reply test-requirements.txt:1]: Because the import is wrong :wink: -> `from unittest import mock`

**Pull request merged**